aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -2,347 +2,120 @@
-            )),
-            SpeculativeStartValue::LastCommittedValue(value) => Ok(*value),
-            SpeculativeStartValue::AggregatedValue(value) => Ok(*value),
-        }
-    }
-
-    pub fn get_value_for_read(&self) -> Result<u128, PanicError> {
-        match self {
-            SpeculativeStartValue::Unset => Err(code_invariant_error(
-                "Tried calling get_value_for_read on Unset speculative value",
-            )),
-            SpeculativeStartValue::LastCommittedValue(_) => Err(code_invariant_error(
-                "Tried calling get_value_for_read on LastCommittedValue speculative value",
-            )),
-            SpeculativeStartValue::AggregatedValue(value) => Ok(*value),
-        }
-    }
-}
-
-/// Describes the state of each aggregator instance.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum AggregatorState {
-    // If aggregator is created in this transaction.
-    Create {
-        value: u128,
-    },
-    Delta {
-        speculative_start_value: SpeculativeStartValue,
-        delta: SignedU128,
-        history: DeltaHistory,
-    },
-}
-
-// Aggregator snapshot is immutable struct, once created - value is fixed.
-// If we want to provide mutability APIs in the future, it should be
-// copy-on-write - i.e. a new aggregator_id should be created for it.
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub enum AggregatorSnapshotState {
-    // Created in this transaction, with explicit value
-    Create {
-        value: SnapshotValue,
-    },
-    // Created in this transaction, via snapshot(&aggregator)
-    Delta {
-        base_aggregator: DelayedFieldID,
-        delta: SignedU128,
-    },
-    // Created in this transaction, via string_concat(prefix, &snapshot, suffix)
-    Derived {
-        base_snapshot: DelayedFieldID,
-        formula: SnapshotToStringFormula,
-    },
-    // Accessed in this transaction, based on the ID
-    Reference {
-        // always expensive/aggregated read
-        speculative_value: SnapshotValue,
-    },
-}
-
-#[derive(Debug)]
-pub struct AggregatorSnapshot {
-    // The identifier used to identify the aggregator.
-    #[allow(dead_code)]
-    id: DelayedFieldID,
-
-    state: AggregatorSnapshotState,
-}
-
-impl AggregatorSnapshot {
-    pub fn into(self) -> AggregatorSnapshotState {
-        self.state
-    }
-}
-
-/// Internal aggregator data structure.
-#[derive(Debug)]
-pub struct Aggregator {
-    // The identifier used to identify the aggregator.
-    id: AggregatorVersionedID,
-    // Describes an upper bound of an aggregator. If value of the aggregator
-    // exceeds it, the aggregator overflows.
-    pub max_value: u128,
-    // Describes a state of an aggregator.
-    pub state: AggregatorState,
-}
+use move_binary_format::errors::PartialVMResult;
+use std::collections::{btree_map::Entry, BTreeMap};
 
 fn get_delayed_field_value_from_storage(
     id: &DelayedFieldID,
     resolver: &dyn DelayedFieldResolver,
-    mode: DelayedFieldReadMode,
```

Comment:
Why is ReadMode removed for Aggregator V2?

Reply:
I've removed a way to be able to get non-aggregated value outside of resolver - and leak non-read-captured information 

last_committed was only used for try_add, and try_add logic is moved to be inside the view.rs, so it can properly capture it's consequences into DeltaHistory inside CapturedReads, and resolver can then only return "true" or "false" there.

also caching of results is moved from extension into CapturedReads instead